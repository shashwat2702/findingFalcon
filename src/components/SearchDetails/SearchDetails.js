/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './SearchDetails.scss';
import DropDown from '../shared/DropDown/DropDown';
import RadioButton from '../shared/RadioButton/RadioButton';

export default class SearchDetails extends Component {
  state={
    showVehicles: false,
  }

  changeVisibilityOfVehicles = (event) => {
    const { target } = event;
    const { value } = target;
    if (value === 'Select One') {
      this.setState({ showVehicles: false });
    } else {
      this.setState({ showVehicles: true });
    }
  }

  onChange = (event) => {
    const { onDropDownChange } = this.props;
    onDropDownChange(event);
    this.changeVisibilityOfVehicles(event);
  }

  render() {
    const {
      planets, vehicles, planetNumber, selectedPlanets, onRadioClick,
    } = this.props;
    const { showVehicles } = this.state;
    return (
      <div className="planetDropDown">
        <h3>
    Planet
          {' '}
          {planetNumber}
        </h3>
        <DropDown
          planets={planets}
          selectedPlanets={selectedPlanets}
          onChange={this.onChange}
          name={planetNumber}
        />
        {showVehicles
          && (
          <RadioButton
            vehicles={vehicles}
            onRadioClick={onRadioClick}
            planetNumber={planetNumber}
          />
          )
        }
      </div>
    );
  }
}

SearchDetails.propTypes = {
  planets: PropTypes.array.isRequired,
  selectedPlanets: PropTypes.array.isRequired,
  vehicles: PropTypes.array.isRequired,
  planetNumber: PropTypes.number.isRequired,
  onDropDownChange: PropTypes.func.isRequired,
  onRadioClick: PropTypes.func.isRequired,
};
