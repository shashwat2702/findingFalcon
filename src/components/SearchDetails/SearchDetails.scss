.planetDropDown {
    width: 300px;
    margin: 1rem;
    h3 {
        color: #C0C0C0;
        margin: 1rem 0;
        font-size: 1.5rem;
    }
}