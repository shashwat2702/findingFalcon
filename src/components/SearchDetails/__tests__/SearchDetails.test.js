import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchDetails from '../SearchDetails';

jest.mock('react-router-dom');
const planets = [{ name: 'PlanetA', distance: 100 }, { name: 'PlanetB', distance: 150 },
  { name: 'PlanetC', distance: 200 }, { name: 'PlanetD', distance: 50 },
  { name: 'PlanetE', distance: 250 }, { name: 'PlanetF', distance: 300 }];
const vehicles = [{ name: 'VehicleA', speed: 10 }, { name: 'VehicleB', speed: 5 },
  { name: 'VehicleC', speed: 20 }, { name: 'VehicleD', speed: 25 },
  { name: 'VehicleE', speed: 50 }, { name: 'VehicleF', speed: 40 }];
const selectedPlanets = ['PlanetC', 'PlanetF', 'PlanetD', 'PlanetA'];
const selectedVehicles = ['VehicleC', 'VehicleD', 'VehicleF', 'VehicleE'];

describe('SearchDetails snapshot', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<SearchDetails
      planets={planets}
      selectedPlanets={selectedPlanets}
      vehicles={vehicles}
      planetNumber={1}
      onDropDownChange={jest.fn()}
      onRadioClick={jest.fn()}
      selectedVehicles={selectedVehicles}
    />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('changeVisibilityOfVehicles', () => {
  it('should set the showVehicles attribute of state to be true for option other than Select One', () => {
    const wrapper = shallow(<SearchDetails
      planets={planets}
      selectedPlanets={selectedPlanets}
      vehicles={vehicles}
      planetNumber={1}
      onDropDownChange={jest.fn()}
      onRadioClick={jest.fn()}
      selectedVehicles={selectedVehicles}
    />);
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 'PlanetC',
      },
    };
    expect(instance.state.showVehicles).toEqual(false);
    instance.changeVisibilityOfVehicles(event);
    expect(instance.state.showVehicles).toEqual(true);
  });
  it('should set the showVehicles attribute of state to be false for option "Select One"', () => {
    const wrapper = shallow(<SearchDetails
      planets={planets}
      selectedPlanets={selectedPlanets}
      vehicles={vehicles}
      planetNumber={1}
      onDropDownChange={jest.fn()}
      onRadioClick={jest.fn()}
      selectedVehicles={selectedVehicles}
    />);
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 'Select One',
      },
    };
    const event2 = {
      target: {
        value: 'PlanetC',
      },
    };
    instance.changeVisibilityOfVehicles(event2);
    expect(instance.state.showVehicles).toEqual(true);
    instance.changeVisibilityOfVehicles(event);
    expect(instance.state.showVehicles).toEqual(false);
  });
});

describe('numberOfOccupiedvehicle', () => {
  it('should return an object having keys as vehicle name and count as value', () => {
    const wrapper = shallow(<SearchDetails
      planets={planets}
      selectedPlanets={selectedPlanets}
      vehicles={vehicles}
      planetNumber={1}
      onDropDownChange={jest.fn()}
      onRadioClick={jest.fn()}
      selectedVehicles={selectedVehicles}
    />);
    const instance = wrapper.instance();
    expect(instance.numberOfOccupiedvehicle(selectedVehicles)).toEqual({
      VehicleC: 1, VehicleD: 1, VehicleE: 1, VehicleF: 1,
    });
  });
});

describe('onChange', () => {
  it('should call on DropDownChange with an argument and should change the visibility of vehicle', () => {
    const onDropDownChangeMock = jest.fn();
    const wrapper = shallow(<SearchDetails
      planets={planets}
      selectedPlanets={selectedPlanets}
      vehicles={vehicles}
      planetNumber={1}
      onDropDownChange={onDropDownChangeMock}
      onRadioClick={jest.fn()}
      selectedVehicles={selectedVehicles}
    />);
    const event = {
      target: {
        value: 'PlanetC',
      },
    };
    const instance = wrapper.instance();
    instance.onChange(event);
    expect(onDropDownChangeMock).toHaveBeenCalledWith(event);
    expect(instance.state.showVehicles).toEqual(true);
  });
});
