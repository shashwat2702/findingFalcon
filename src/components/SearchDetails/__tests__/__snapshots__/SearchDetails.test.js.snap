// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDetails snapshot should render correctly 1`] = `
<div
  className="planetDropDown"
>
  <h3>
    Planet
     
    1
  </h3>
  <div
    className="dropDown"
  >
    <select
      name={1}
      onChange={[Function]}
      value="PlanetC"
    >
      <option
        value=""
      >
        Select One
      </option>
      <option
        value="PlanetC"
      >
        PlanetC
      </option>
      <option
        value="PlanetB"
      >
        PlanetB
      </option>
      <option
        value="PlanetE"
      >
        PlanetE
      </option>
    </select>
  </div>
</div>
`;
